src/scan.ts

```diff
@@ -171,7 +204,10 @@ async function run() {
     await exec.exec(`endorctl`, ["scan", "--path=.", ...options], scanOptions);
 
     core.info("Scan completed successfully!");
-    core.setOutput("result", scanResult);
+
+    if (EXPORT_SCAN_RESULT_ARTIFACT && SCAN_SUMMARY_OUTPUT_TYPE === "json") {
+      uploadArtifact(scanResult);
+    }
```

Comment:
minor: the result of `uploadArtifact` could be awaited here, to ensure that the result is written and uploaded before the action completes (could also trigger `core.setFailed(...)` on upload failure if desired.

Additional note: if an endorctl scan results has no resulting findings, this does upload an empty artifact. If that behavior is not desired, an additional check for empty `scanResult` could be added in `uploadArtifact`

Reply:
Added the `await` and empty check for `scanResult`.
In case of upload failure, an error annotation is emitted: `core.error("Some items failed to export");`